pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -208,7 +207,13 @@ func (gc *GarbageCollector) Sync(discoveryClient discovery.DiscoveryInterface, p
 		}
 		if !controller.WaitForCacheSync("garbage collector", stopCh, gc.dependencyGraphBuilder.IsSynced) {
 			utilruntime.HandleError(fmt.Errorf("timed out waiting for dependency graph builder sync during GC sync"))
+			return
 		}
+
+		// Finally, keep track of our new state. Do this after all preceding steps
```

Comment:
add a V(2) log message here that sync completed... want to be able to pair with the log message from line 183 to know resync completed

Reply:
Done